src/UI/Implementation/Component/ComponentHelper.php

```diff
@@ -9,6 +9,23 @@
  */
 trait ComponentHelper {
 	/**
+	 * Default implementation uses the namespace of the component up to and excluding
+	 * "Component", reverses the order and adds spaces.
+	 *
+	 * @return string
+	 */
+	public function getName() {
+		$cls = explode("\\", get_class($this));
+		$name = [];
+		$cur = array_pop($cls);
+		while ($cur !== "Component" && count($cls) > 0) {
+			$name[] = preg_replace("%([a-z])([A-Z])%", "$1 $2", $cur);
+			$cur = array_pop($cls);
+		}
```

Comment:
Is it really required to separate camel case classnames with spaces? If not, we could get rid of the regex and simplify to something like:
```php
$name = array_reverse(array_splice(explode("\\", get_class($this)), 1))
```